### PR TITLE
Fix log api - pass context in api calls

### PR DESCRIPTION
**Pull request by Martin2112**
Make the handler code call the context setup function so request handlers have access to values set on it.

### CODE REVIEW

File: trillian/ctfe/requestlog.go [hunk 1]
```diff
@@ -73,56 +73,57 @@ type DefaultRequestLog struct {
...
 // Start logs the start of request processing.
-func (dlr *DefaultRequestLog) Start(ctx context.Context) {
+func (dlr *DefaultRequestLog) Start(ctx context.Context) context.Context {
 	glog.V(vLevel).Info("RL: Start")
+	return ctx
 }
 
 // LogPrefix logs the prefix of the CT log that this request is for.
-func (dlr *DefaultRequestLog) LogPrefix(p string) {
+func (dlr *DefaultRequestLog) LogPrefix(c context.Context, p string) {
```
Comment (daviddrysdale) [inline trillian/ctfe/requestlog.go]: Context parameters are normally named `ctx` -- any reason not to stick to that here (and throughout)?

File: trillian/ctfe/requestlog.go [hunk 2]
```diff
@@ -34,37 +34,37 @@ const vLevel = 9
 type RequestLog interface {
 	// Start will be called once at the beginning of handling each request.
 	// The supplied context will be the one used for request processing and
-	// can be used by the logger to set values.
-	Start(context.Context)
+	// can be used by the logger to set values on the returned context.
```
Comment (daviddrysdale) [inline trillian/ctfe/requestlog.go]: Does that mean that the returned context should be used in all the calls below?  If so, best to say so.

File: trillian/ctfe/requestlog.go (hunk 1 above)
Comment (daviddrysdale) [reply trillian/ctfe/requestlog.go]: (or alternatively use `_` as this incarnation of the interface doesn't use the param)

File: trillian/ctfe/requestlog.go (hunk 2 above)
Comment (Martin2112) [reply trillian/ctfe/requestlog.go]: It's set up by the handler code but I've added an explicit comment.

File: trillian/ctfe/requestlog.go (hunk 1 above)
Comment (Martin2112) [reply trillian/ctfe/requestlog.go]: I've gone with '_' as this impl doesn't keep state in the context.

**Pull request merged**